### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by oscarbc96**
Refactor and bump to version 0.4.0

### CODE REVIEW

File: pycfmodel/model/cf_model.py [hunk 1]
```diff
@@ -22,31 +22,26 @@ def __init__(self, cf_script):
         self.aws_template_format_version = cf_script.get("AWSTemplateFormatVersion")
         self.description = cf_script.get("Description")
         self.metadata = cf_script.get("Metadata")
+
```
Comment (IgnacioRV) [inline pycfmodel/model/cf_model.py]: Is this space separating something in particular or can it be removed?

Comment (oscarbc96) [reply pycfmodel/model/cf_model.py]: It's to separate variables that store info about the CF, from the ones that store the information created by the cloudformation.

File: pycfmodel/model/resources/iam_user.py [hunk 2]
```diff
@@ -42,3 +42,16 @@ def __init__(self, logical_id, value):
         self.set_properties(
             value.get("Properties", {})
         )
+
+    def has_hardcoded_credentials(self):
```
[the remark this answers is not in the document]
Comment (oscarbc96) [reply pycfmodel/model/resources/iam_user.py]: ohh, true

File: pycfmodel/model/resources/properties/policy_document.py [hunk 3]
```diff
@@ -12,141 +12,142 @@
...
-        "IAM:GetRolePolicy",
-        "IAM:DeleteInstanceProfile",
-        "IAM:CreateLoginProfile",
-        "IAM:SimulatePrincipalPolicy",
-        "IAM:UpdateGroup",
-        "IAM:GetPolicy"
-    ]
+from .statement import Statement
 
+_IAM_ACTIONS = [
```
[the remark this answers is not in the document]
Comment (jsoucheiron) [reply pycfmodel/model/resources/properties/policy_document.py]: 👍

File: pycfmodel/model/cf_model.py [hunk 4]
```diff
@@ -12,41 +12,34 @@
...
+    def _parse_resources(self, template_resources):
         """Parses and sets resources in the model using a factory."""
-
-        self.resources = {}
-        resource_factory = ResourceFactory()
-        for res_id, res_value in resources.items():
-            r = resource_factory.create_resource(res_id, res_value)
+        resources = defaultdict(list)
+        for res_id, res_value in template_resources.items():
+            r = create_resource(res_id, res_value)
```
[the remark this answers is not in the document]
Comment (jsoucheiron) [reply pycfmodel/model/cf_model.py]: Good catch

**Pull request merged**